src/LibraryManager/Utilities/FileGlobbingUtility.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Minimatch;
+
+namespace Microsoft.Web.LibraryManager.Utilities
+{
+    internal class FileGlobbingUtility
+    {
+        public static IEnumerable<string> ExpandFileGlobs(IEnumerable<string> potentialGlobs, IEnumerable<string> libraryFiles)
+        {
+            var finalSetOfFiles = new HashSet<string>();
+            var negatedOptions = new Minimatch.Options { FlipNegate = true };
+
+            foreach (string potentialGlob in potentialGlobs)
+            {
+                // only process globs where we find them, otherwise it can get expensive
+                if (potentialGlob.StartsWith("!", StringComparison.Ordinal))
+                {
+                    // Remove matches from the files list
+                    var filesToRemove = finalSetOfFiles.Where(f => Minimatcher.Check(f, potentialGlob, negatedOptions)).ToList();
+                    foreach (string file in filesToRemove)
+                    {
+                        finalSetOfFiles.Remove(file);
+                    }
+                }
+                else if (potentialGlob.Contains("*"))
```

Comment:
This isn't looking for disallowed characters (! is a valid file name character on Windows, and invalid path characters are [mostly a Windows-only limitation](https://stackoverflow.com/questions/1311037/are-there-any-invalid-linux-filenames)).  This is just looking for characters indicating that this is a file pattern, so that we know whether to attempt to expand it or to preserve it as a literal file name.  `?` would make sense to add here, as well as `[`.  There's some more esoteric patterns that minimatch also supports, but I don't know if they'd actually get used so I didn't look for them here.